### PR TITLE
Update `stylelint` & `stylelint-config-wagtail` packages, adopt a subset of new rules

### LINKED ISSUE #10719: Update Stylelint to v15* & update stylelint-config-wagtail
This is a maintenance task to update our Stylelint packages to 15, alongside the related linting packages we use such as stylelint-config-wagtail.

It would be good to align with the new rules that come with our shared Stylelint config also. 

## Links

- https://stylelint.io/migration-guide/to-15/
- https://github.com/wagtail/wagtail/pull/10650
- https://github.com/wagtail/wagtail/pull/10671
- https://github.com/wagtail/stylelint-config-wagtail/issues/33

### PR DESCRIPTION
**Pull request by lb-**
## Overview

Closes #10719

This PR does not aim to align 100% with the new `stylelint-config-wagtail` updates but will ensure we do not get a [huge amount of warnings](https://github.com/wagtail/wagtail/issues/10719#issuecomment-1751844763) when running `npm install`.

Instead, the goal is to adopt any auto 'fixing' of our styles files and a subset of easy to adopt rules without this PR being a huge item to review.

This also means that new styles added can align with the rules as they are in place and we avoid adding more complex styles than we already have (i.e. in the area of `selector-max-combinators` or `selector-max-specificity`.

## Implementation notes

- Ensure that some rules are always ignored in overrides (aka vendor) styles. This avoids us having to add ignore comments all over these styles.
- Formatting & ordering (auto applied), this includes the padding/inset shorthand changes
- Adopt a small set of no-union-classes changes as an example of changes to come in future PRs
- Ignore some areas where we are selecting against data* attributes that will not be practical to change
- Move some no-important ignore rules to specific lines
- Ignore max-combinators in modeladmin styles (legacy)
- Remove error messages forced-color-adjust setting to none

## Testing notes

There is only one functional change: Messages in high contrast mode (with and without dark mode).

## Future work

There are potentially 100s of additional changes to make to fully align with the stylelint-config-wagtail base styles.

Once merged, I will look at raising some issues that help us get incremental steps towards alignment, these will be potentially good first issues (for ones that are just renames of classes for example).

### CODE REVIEW

**Comment by lb-** [comment]
Need to manually rebase.

File: .stylelintrc.js [hunk 1]
```diff
@@ -79,5 +86,25 @@ module.exports = {
...
   },
+  overrides: [
+    // Always ignore some rules in vendor overrides as we have no control over the way these are written.
+    {
+      files: ['client/scss/overrides/**/*.scss'],
+      rules: {
+        'max-nesting-depth': 5,
+        'selector-attribute-name-disallowed-list': null,
+        'selector-class-pattern': null,
+        'selector-max-combinators': 5,
```
Comment (thibaudcolas) [inline .stylelintrc.js:106]: I see only two of those rules currently raising issues, and only in two files from our overrides, so I think we’d be better off with in-file overrides here.

**Review by thibaudcolas — APPROVED** [review]
Thank you @lb-! I’d like to further discuss `declaration-block-no-redundant-longhand-properties` before we enforce it in Wagtail so I’ve taken those changes out for now, I hope that’s ok.

I’ve also moved the overrides to `overrides/**` to be directly inside the files – as there are fewer than what the config suggests.

Also worth noting this includes a Sass update which introduces deprecation warnings (one related to shorthand properties which I took out, one which I’ll address separately). When we update our dependencies via transitive updates like this, I think it’d be nice to keep the `package.json` up-to-date – so I’ve done that.

This doesn’t change which version is installed, just avoids a bit of confusion if our package.json says `^1.0.0` or similar but actually it’s resolving to `1.69.5`.

**Comment by thibaudcolas** [comment]
I’ve opened https://github.com/wagtail/stylelint-config-wagtail/issues/41 and https://github.com/wagtail/stylelint-config-wagtail/issues/42 as possible follow-ups.

File: .stylelintrc.js (hunk 1 above)
Comment (lb-) [reply .stylelintrc.js]: Good call thank you.

**Comment by lb-** [comment]
Thanks @thibaudcolas - I agree with the shorthand properties thing. From the outside it seems like a helpful feature but it does make changing code harder. Thanks for reviewing and resolving those items.

**Pull request merged**